curriculum/src/build-curriculum.ts

```diff
@@ -335,9 +335,125 @@ export async function parseCurriculumStructure(filter?: Filter) {
   };
 }
 
-export async function buildCurriculum(lang: string, filters?: Filter) {
-  const contentDir = getContentDir(lang);
+// When in selective mode, we need to ensure that GraphQL is satisfied with basic query data,
+// so we build a minimal full-stack-developer challenge if it wasn't included above.
+async function ensureGraphQLSchema(
```

Comment:
We use [createSchemaCustomization](https://github.com/freeCodeCamp/freeCodeCamp/blob/main/client/gatsby-node.js#L295) to define the expected schema. It should take care of that and, if it isn't, building a single challenge doesn't guarantee that Gatsby will be able to build. Reason being: it can't build if it tries a GraphQL query for data that neither exists nor is specified by `createSchemaCustomization`. A single challenge won't include all the data, so is not always going to fix the issue.